eternalcore-core/src/main/java/com/eternalcode/core/feature/teleport/TeleportTask.java

```diff
@@ -40,54 +38,76 @@ class TeleportTask implements Runnable {
 
     @Override
     public void run() {
-        List<Teleport> teleportsCopy = new ArrayList<>(this.teleportTaskService.getTeleports());
-
-        for (Teleport teleport : teleportsCopy) {
-            Location destinationLocation = PositionAdapter.convert(teleport.getDestinationLocation());
-            Location startLocation = PositionAdapter.convert(teleport.getStartLocation());
-            UUID uuid = teleport.getUuid();
-            Instant teleportMoment = teleport.getTeleportMoment();
+        for (Teleport teleport : this.teleportTaskService.getTeleports()) {
+            UUID uuid = teleport.getPlayerUniqueId();
 
             Player player = this.server.getPlayer(uuid);
 
             if (player == null) {
+                this.teleportTaskService.removeTeleport(uuid);
                 continue;
             }
 
-            if (player.getLocation().distance(startLocation) > 0.5) {
-                this.teleportTaskService.removeTeleport(uuid);
-
-                this.noticeService.create()
-                    .notice(translation -> Notice.actionbar(StringUtils.EMPTY))
-                    .notice(translation -> translation.teleport().teleportTaskCanceled())
-                    .player(player.getUniqueId())
-                    .send();
-
+            if (this.isPlayerMovedDuringTeleport(player, teleport) || this.checkTeleportTimer(player, teleport)) {
```

Comment:
`this.isPlayerMovedDuringTeleport(player, teleport)` -> `this.hasPlayerMovedDuringTeleport(player, teleport)`
we should use past tense or present continuous tense `this.isPlayerMovingDuringTeleport(player, teleport)`